packages/gatsby-plugin-page-creator/src/extract-query.ts

```diff
@@ -12,15 +12,20 @@ import {
 // End result should be something like { allProducts { nodes { id }}}
 export function generateQueryFromString(
   queryOrModel: string,
-  fileAbsolutePath: string
+  fileAbsolutePath: string,
+  nodeIds?: Array<string>
 ): string {
   // TODO: 'fields' possibly contains duplicate fields, e.g. field{name},field{description} that should be merged to field{name,description}
   const fields = extractUrlParamsForQuery(fileAbsolutePath)
 
   // In case queryOrModel is not capitalized
   const connectionQuery = _.camelCase(`all ${queryOrModel}`)
 
-  return `{${connectionQuery}{nodes{${fields}}}}`
+  const connectionArgs = nodeIds
```

Comment:
A simple unit test case for this would be cool. The existing unit tests are also good to understand how things work via their inline snapshots

Reply:
added in https://github.com/gatsbyjs/gatsby/pull/36623/commits/11cc6cba8b2568231f7d07d2e0237158fe5e3001 - it is single simple test that I copied from one of existing only adding those new optional `nodeIds` in with snapshot showing the filter